datafusion/src/physical_plan/file_format/parquet.rs

```diff
@@ -385,9 +389,33 @@ fn build_row_group_predicate(
     }
 }
 
+// Map projections from the schema which merges all file schemas to projections on a particular
+// file
+fn map_projections(
+    merged_schema: &Schema,
+    file_schema: &Schema,
+    projections: &[usize],
+) -> Result<Vec<usize>> {
+    let mut mapped: Vec<usize> = vec![];
+    for idx in projections {
+        let field = merged_schema.field(*idx);
+        if let Ok(mapped_idx) = file_schema.index_of(field.name().as_str()) {
+            if file_schema.field(mapped_idx).data_type() == field.data_type() {
+                mapped.push(mapped_idx)
+            } else {
+                let msg = format!("Failed to map column projection for field {}. Incompatible data types {:?} and {:?}", field.name(), file_schema.field(mapped_idx).data_type(), field.data_type());
+                error!("{}", msg);
+                return Err(DataFusionError::ParquetError(ParquetError::General(msg)));
```

Comment:
I recommend
1. Using a `info` message (as depending on the usecase this may signal a user error rather than something to investigate)
2. Using a `DataFusionError` rather than one from Parquet (the rationale being that this error comes from DataFusion, and is not related to being able to read the parquet files)

```suggestion
                let msg = format!("Failed to map column projection for field {}. Incompatible data types {:?} and {:?}", field.name(), file_schema.field(mapped_idx).data_type(), field.data_type());
                info!("{}", msg);
                return Err(DataFusionError::Execution(msg));
```